### ISSUE TITLE
Unable to compile on El Capitan on case sensitive file system

### ISSUE THREAD

**Issue by wmalloc** [danthorpe/Operations#200]
CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler
(5 failures)
/Users/malik/Library/Developer/Xcode/DerivedData/Permissions-dhbmprznmjtvyugxqnfdekwzvzfs/Build/Intermediates/Pods.build/Release-iphoneos/Operations.build/unextended-module.modulemap:2:19: error: umbrella header 'Operations-umbrella.h' not found
<unknown>:0: error: could not build Objective-C module 'Operations'
/Users/malik/Library/Developer/Xcode/DerivedData/Permissions-dhbmprznmjtvyugxqnfdekwzvzfs/Build/Intermediates/Pods.build/Release-iphoneos/Operations.build/unextended-module.modulemap:2:19: error: umbrella header 'Operations-umbrella.h' not found
<unknown>:0: error: could not build Objective-C module 'Operations'
/Users/malik/Library/Developer/Xcode/DerivedData/Permissions-dhbmprznmjtvyugxqnfdekwzvzfs/Build/Intermediates/Pods.build/Release-iphoneos/Operations.build/unextended-module.modulemap:2:19: error: umbrella header 'Operations-umbrella.h' not found
<unknown>:0: error: could not build Objective-C module 'Operations'
/Users/malik/Library/Developer/Xcode/DerivedData/Permissions-dhbmprznmjtvyugxqnfdekwzvzfs/Build/Intermediates/Pods.build/Release-iphoneos/Operations.build/unextended-module.modulemap:2:19: error: umbrella header 'Operations-umbrella.h' not found
<unknown>:0: error: could not build Objective-C module 'Operations'
A shell task failed with exit code 65:

**Comment by wmalloc** [comment]
Xcode Version 7.2.1 (7C1002)

**Comment by danthorpe** [comment]
Did you run pod install?

**Comment by wmalloc** [comment]
That is what is confusing is that I am not using it as a Pod, I am using with carthage, and when I do carthage update --platform iOS above output is given, not sure why it is trying to deal with pod header files.

If I do the same thing without the case sensitive file system the carthage update succeeds.

**Comment by danthorpe** [comment]
Okay, I think I know what is going on.

So, is this the scenario? You have a project, using Carthage, which you're using with `github "danthorpe/Operations"` in the Cartfile.

And you're using the latest version of Carthage? Because in this version, I think Carthage will go off an attempt to build *every* .xcodeproj that it can find - e.g. the example app Permissions, which itself uses CocoaPods to setup the dependency on Operations.

Ultimately, I think this is a bug in Carthage - as it prevents having additional projects in repos. I'll take a look into it.

Thanks for the feedback on this - the error is pretty arcane!

**Comment by wmalloc** [comment]
I am using the latest release 0.14.0. I guess I should let Carthage folks know what is going on.

**Comment by danthorpe** [comment]
I was gonna say - hold off on that for now. I'll try to investigate this tonight.

**Comment by danthorpe** [comment]
Also just to be clear - I don't see any evidence that this is in any way related to a case sensitive file system - why do you think this is the case?

**Comment by wmalloc** [comment]
I have setup the same on a case insensitive system and it builds without any issues, They are both setup
El Capitan (latest release version) (one system with case sensitive and one case insensitive)
Xcode
Carthage 0.14.0
CocoaPods 0.39.0

This is by process of elimination. I can try another system and try to dig in some more

**Comment by danthorpe** [comment]
Hmm, Okay. That's weird. I've got to leave this right now - will come back to it tomorrow.

**Comment by danthorpe** [comment]
Just to clarify, using the latest version of Carthage, on a project using Operations I get this:

```
*** Building scheme "OS X Operations" in Operations.xcodeproj
*** Building scheme "tvOS Operations" in Operations.xcodeproj
*** Building scheme "watchOS Operations" in Operations.xcodeproj
*** Building scheme "iOS Operations" in Operations.xcodeproj
*** Building scheme "OS X Extensions Operations" in Operations_ExtensionCompatible.xcodeproj
*** Building scheme "tvOS Extensions Operations" in Operations_ExtensionCompatible.xcodeproj
*** Building scheme "iOS Extensions Operations" in Operations_ExtensionCompatible.xcodeproj
*** Building scheme "Operations" in Permissions.xcworkspace
** BUILD FAILED **
```

I don't think it has anything to do with a case insensitive file system. (I don't have a case insensitive file system).

**Comment by danthorpe** [comment]
@wmalloc can you check on your case-sensitive file system if build Operations using carthage is still an issue. Please try building from `development`

```
github "danthorpe/Operations" "development"
```

I think I've fixed this temporarily.

**Comment by wmalloc** [comment]
Thanks @danthorpe the development branch compiled correctly.

*** Building scheme "iOS Operations" in Operations.xcodeproj
*** Building scheme "iOS Extensions Operations" in Operations_ExtensionCompatible.xcodeproj
*** Building scheme "Operations" in Last Opened.xcworkspace
*** Building scheme "Operations" in Last Opened.xcworkspace

**Comment by danthorpe** [comment]
Okay, great. I'm gonna close this for now. I have an issue in Carthage to figure out how it should ultimately work with example projects.

**Comment by briancordanyoung** [comment]
@wmalloc @danthorpe What is the best way to make sure I'm using this version.  A Cartfile like this?:
github "danthorpe/Operations" "release/2.6.1"

**Comment by wmalloc** [comment]
@briancordanyoung if you do not specify any version number you will get the latest release version which is 2.6.1.

But I tested this version:

github "danthorpe/Operations" "development"

**Comment by briancordanyoung** [comment]
@wmalloc I tried that and got 2.6.0   I run:
> carthage update --platform iOS
After updating the Cartfile... yes?

**Comment by briancordanyoung** [comment]
@wmalloc 2.6.0 is still marked as the latest release, fro what I can tell. [releases](https://github.com/danthorpe/Operations/releases)

**Comment by danthorpe** [comment]
@briancordanyoung your first comment should work for you. I have picked that fix into the 'release/2.6.1' branch. This is a release candidate - but it will still report the version as 2.6.0 as the only change is this fix. No version number bumps.

However I will be finishing this off in a few hours - so then you shouldn't need to change anything.
